ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ScanConfigurationSelect.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useContext, useState } from 'react';
+import {
+    Divider,
+    MenuToggle,
+    MenuToggleElement,
+    Select,
+    SelectGroup,
+    SelectList,
+    SelectOption,
+    Spinner,
+} from '@patternfly/react-core';
+
+import { ScanConfigurationsContext } from '../ScanConfigurationsProvider';
+
+const ALL_SCAN_SCHEDULES_OPTION = 'All scan schedules';
+
+function ScanConfigurationSelect() {
+    const [isOpen, setIsOpen] = useState(false);
+
+    const { scanConfigurationsQuery, selectedScanConfig, setSelectedScanConfig } =
+        useContext(ScanConfigurationsContext);
+
+    const onToggleClick = () => {
+        setIsOpen(!isOpen);
+    };
+
+    const onSelect = (
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        value: string | number | undefined
+    ) => {
+        const selectedValue = value === ALL_SCAN_SCHEDULES_OPTION ? undefined : (value as string);
+        setSelectedScanConfig(selectedValue);
+        setIsOpen(false);
+    };
+
+    const renderToggle = (toggleRef: React.Ref<HTMLButtonElement | MenuToggleElement>) => {
+        return (
+            <MenuToggle ref={toggleRef} onClick={onToggleClick} isExpanded={isOpen}>
+                {(selectedScanConfig as string) || ALL_SCAN_SCHEDULES_OPTION}
+            </MenuToggle>
+        );
+    };
+
+    return (
+        <Select
+            id="scan-schedules-filter-id"
+            isOpen={isOpen}
+            selected={selectedScanConfig || ALL_SCAN_SCHEDULES_OPTION}
+            onSelect={onSelect}
+            onOpenChange={(isOpen) => setIsOpen(isOpen)}
+            toggle={renderToggle}
+            shouldFocusToggleOnSelect
+        >
+            {scanConfigurationsQuery.isLoading ? (
+                <SelectOption isLoading value="loader">
+                    <Spinner size="lg" />
+                </SelectOption>
```

Comment:
At this moment, my mind is not quite up to the challenge for a careful review of this conditional rendering.
* Bravo for handling `isLoading`
* What are pro and con of `isDisabled` prop
    https://www.patternfly.org/components/menus/select#single-select

Reply:
In case it is relevant, here is conditional code to display a selected option waiting for a response:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Components/NotifierConfiguration/NotifierMailingLists.tsx#L91-L103
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Components/NotifierConfiguration/NotifierMailingLists.tsx#L118

Reply:
My thought process was if the call hangs at all the loading indicator might be more intuitive than just disabling the select.


https://github.com/stackrox/stackrox/assets/61400697/154832f8-5b77-46b0-8d05-fe64b4fd3480



The question made me do some additional testing and made me realize I need to handle a non-existent scan schedule better.

Reply:
That also makes me wonder how to handle an error.. This is what currently happen. Not sure it warrants an alert but that would be consistent with how we typically handle errors as far as I know.

![Screenshot 2024-06-21 at 11 10 21 AM](https://github.com/stackrox/stackrox/assets/61400697/d48b8dbd-dd5e-4598-9f64-7e80faa531a0)

Reply:
This sounds like a good question for team meeting: with limited space for advanced filter, how does vulnerabilities handle request errors?